### PR TITLE
Store VersionMaps in SQLite

### PR DESCRIPTION
**Pull request by csilvestrini**
VersionMaps are serialized to proto, which are then serialized to a hex
string, which are stored in the DB as a TEXT column. A BLOB would be
more typical, but the Android SQLite API doesn't handle BLOBs properly.

Diffbase #4681

### CODE REVIEW

File: java/arcs/android/crdt/ParcelableVersionMap.kt [hunk 1]
```diff
@@ -26,4 +25,4 @@ fun VersionMap.toProto(): VersionMapProto = VersionMapProto.newBuilder()
 
 /** Reads a [VersionMap] out of a [Parcel]. */
 fun Parcel.readVersionMap(): VersionMap? =
-    readProto(VersionMapProto.getDefaultInstance())?.let { VersionMap(it) }
+    readProto(VersionMapProto.getDefaultInstance())?.let { fromProto(it) }
```
Comment (jasonwyatt) [inline java/arcs/android/crdt/ParcelableVersionMap.kt]: Should this file be renamed to `ProtoVersionMap.kt`?

File: java/arcs/android/storage/database/DatabaseImpl.kt [hunk 2]
```diff
@@ -1052,7 +1120,9 @@ class DatabaseImpl(
                 -- Maps entity storage key IDs to entity IDs (Arcs string ID, not a row ID).
                 CREATE TABLE entities (
                     storage_key_id INTEGER NOT NULL PRIMARY KEY,
-                    entity_id TEXT NOT NULL
+                    entity_id TEXT NOT NULL,
+                    -- Serialized VersionMapProto for the entity.
+                    version_map TEXT NOT NULL
```
Comment (jasonwyatt) [inline java/arcs/android/storage/database/DatabaseImpl.kt]: Curious why you opted for a text column, as opposed to a BLOB?

Comment (jasonwyatt) [reply java/arcs/android/storage/database/DatabaseImpl.kt]: Ah lol. I see the PR description now. :man_facepalming:

File: java/arcs/android/crdt/ParcelableVersionMap.kt (hunk 1 above)
Comment (csilvestrini) [reply java/arcs/android/crdt/ParcelableVersionMap.kt]: Done (went with `VersionMapProto.kt` -- named the same as the proto itself)

**Comment by csilvestrini** [comment]
Presubmit passing in cl/296100593 (failure is unrelated)

**Pull request merged**